apps/wallet/src/stores/station.store.ts

```diff
@@ -419,15 +426,46 @@ export const useStationStore = defineStore('station', {
           throw new Error(`Invalid next version response, expected WasmModule`);
         }
 
+        let nextUpgraderVersion: string | undefined;
         this.versionManagement.nextStationVersion = registryEntry.value.WasmModule.version;
         for (const dependency of registryEntry.value.WasmModule.dependencies) {
           if (dependency.name === '@orbit/upgrader' && dependency.version !== upgraderVersion) {
-            this.versionManagement.nextUpgraderVersion = dependency.version;
+            nextUpgraderVersion = dependency.version;
             break;
           }
         }
+
+        this.versionManagement.nextUpgraderVersion = nextUpgraderVersion;
+
+        // check if there is an existing request to update the canister, if so, we store the request id
+        // to avoid prompting the user to update again if they have already requested an update
+        this.service
```

Comment:
Style nit: this could also be converted to async/await syntax for consistency.

Reply:
Did that for the success scenario now, i am still using the catch there though, main reason is that if that call fails i don't want to throw to the overall try/catch block to prevent breaking the new version verification flow, also added a comment about that.

Reply:
A nested try/catch will have the same effect as using a `.catch` call nested inside a try/catch. Either way is fine for me anyway 👍